### PR TITLE
bpo-35727: Use exit code 0 on sys.exit() in multiprocessing.Process.

### PR DESCRIPTION
**Pull request by chrahunt**
https://bugs.python.org/issue35727

### CODE REVIEW

File: Lib/multiprocessing/process.py
Comment (eamanu) [inline Lib/multiprocessing/process.py:324]: Thinking  this exitcode. Why we need return a '1'?

File: Lib/test/_test_multiprocessing.py
Comment (eamanu) [inline Lib/test/_test_multiprocessing.py]: I would like add an integer here

File: Lib/multiprocessing/process.py
Comment (chrahunt) [reply Lib/multiprocessing/process.py:324]: The behavior aligns with the description of [`sys.exit`](https://docs.python.org/3/library/sys.html#sys.exit).

>  If another type of object is passed, `None` is equivalent to passing zero, and any other object is printed to `stderr` and results in an exit code of 1.

File: Lib/test/_test_multiprocessing.py
Comment (chrahunt) [reply Lib/test/_test_multiprocessing.py]: The test for the integer case should be covered earlier in the test case [here](https://github.com/python/cpython/pull/11538/files/36f0c9bc1990cc9507626f96f71caade59e274b8#diff-b046ab474480855fb4a01de88cfc82bbR778). I can combine both sets of cases.

File: Lib/multiprocessing/process.py
Comment (eamanu) [reply Lib/multiprocessing/process.py:324]: Ok, thanks.

**Review by eamanu — APPROVED** [review]
LGTM. I will wait for review from someone else. Thanks!

File: Lib/test/_test_multiprocessing.py
Comment (asottile) [inline Lib/test/_test_multiprocessing.py:859]: can probably write these as [subtests](https://docs.python.org/3/library/unittest.html#distinguishing-test-iterations-using-subtests)

also probably add a case for the non-int, non-None case

Comment (chrahunt) [reply Lib/test/_test_multiprocessing.py:859]: Updated to use `self.subTest`. The non-int, non-None cases are covered by the set of tests above (`[1, 2, 3]`, `'ignore this'`).

**Comment by applio** [comment]
Added a question in issue35727 on bpo.
At a minimum, this patch will need to be updated to reflect more recent changes (mostly from June, I think) but we should discuss more on bpo.

**Comment by chrahunt** [comment]
@applio, thanks for taking a look! I rebased on master (it didn't require any changes I could see) and commented on your points on bpo.

File: Lib/test/_test_multiprocessing.py
Comment (pitrou) [inline Lib/test/_test_multiprocessing.py:862]: For the record, why did you have to set this process to daemon?

**Review by pitrou — APPROVED** [review]
LGTM, just one question though

File: Lib/test/_test_multiprocessing.py
Comment (pitrou) [reply Lib/test/_test_multiprocessing.py:862]: I see, it's simply a leftover of the previous test case.

**Pull request merged**
